ethcore/light/src/on_demand/mod.rs

```diff
@@ -19,44 +19,54 @@
 //! will take the raw data received here and extract meaningful results from it.
 
 use std::cmp;
-use std::collections::{HashMap, BTreeSet};
+use std::collections::HashMap;
 use std::marker::PhantomData;
 use std::sync::Arc;
+use std::time::Duration;
 
 use ethcore::executed::{Executed, ExecutionError};
-
 use futures::{Poll, Future, Async};
 use futures::sync::oneshot::{self, Receiver};
 use network::PeerId;
 use parking_lot::{RwLock, Mutex};
 use rand;
-use std::time::{Duration, SystemTime};
+use rand::Rng;
 
 use net::{
-	self, Handler, PeerStatus, Status, Capabilities,
+	Handler, PeerStatus, Status, Capabilities,
 	Announcement, EventContext, BasicContext, ReqId,
 };
+
 use cache::Cache;
 use request::{self as basic_request, Request as NetworkRequest};
 use self::request::CheckedRequest;
 
-pub use self::request::{Request, Response, HeaderRef};
+pub use self::request::{Request, Response, HeaderRef, Error as ValidityError};
+pub use self::request_guard::{RequestGuard, Error as RequestError};
+pub use self::response_guard::{ResponseGuard, Error as ResponseGuardError, Inner as ResponseGuardInner};
+
+pub use types::request::ResponseError;
 
 #[cfg(test)]
 mod tests;
 
 pub mod request;
+mod request_guard;
+mod response_guard;
 
 /// The result of execution
 pub type ExecutionResult = Result<Executed, ExecutionError>;
 
-/// The default number of retries for OnDemand queries to send to the other nodes
-pub const DEFAULT_RETRY_COUNT: usize = 10;
-
-/// The default time limit in milliseconds for inactive (no new peer to connect to) OnDemand queries (0 for unlimited)
-pub const DEFAULT_QUERY_TIME_LIMIT: Duration = Duration::from_millis(10000);
-
-const NULL_DURATION: Duration = Duration::from_secs(0);
+/// The initial backoff interval for OnDemand queries
+pub const DEFAULT_REQUEST_MIN_BACKOFF_DURATION: Duration = Duration::from_secs(10);
+/// The maximum request interval for OnDemand queries
+pub const DEFAULT_REQUEST_MAX_BACKOFF_DURATION: Duration = Duration::from_secs(100);
+/// The default window length a response is evaluated
+pub const DEFAULT_RESPONSE_TIME_TO_LIVE: Duration = Duration::from_secs(60);
```

Comment:
Why was it changed from 10 secs to 1min, isn't 1min too long for a user to wait?

Reply:
I think we still need to (but haven't) revert this line at this moment, right?

Reply:
Yeah, I should have commented instead of approving the PR to avoid the merge before fixing the comment, sorry.